uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/LineChartPopupMediaItem.cpp

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+ *  Copyright 2012-2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "LineChartPopupMediaItem.h"
+
+// Qt headers
+#include <QPointF>
+
+// Maps SDK headers
+#include <PopupMedia.h>
+#include <PopupMediaValue.h>
+
+// Toolkit headers
+#include <PopupMediaItem.h>
+
+namespace Esri::ArcGISRuntime::Toolkit {
+
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::LineChartPopupMediaItem
+  \inmodule ArcGISRuntimeToolkit
+  \brief Represents the contents of a \c{MediaPopupElement::media} to be used for a LineSeries in QML.
+  */
+
+/*!
+  \brief Constructor
+  \list
+    \li \a popupMedia - The \l {Esri::ArcGISRuntime::PopupMedia} {PopupMedia} used to populate the list of points.
+    \li \a jsonColorArray - The JSON array containing the color information for the line.
+    \li \a parent - The optional parent QObject.
+  \endlist
+ */
+LineChartPopupMediaItem::LineChartPopupMediaItem(PopupMedia* popupMedia, const QJsonArray& jsonColorArray, QObject* parent)
+  : PopupMediaItem{popupMedia, parent},
+    m_jsonColorArr{jsonColorArray}
+{
+  auto mediaValue = popupMediaItem()->value();
+  const auto popupMediaValueDataLength = mediaValue->data().length();
+
+  for (int i = 0; i < popupMediaValueDataLength; i++ )
+  {
+    const auto label = mediaValue->labels().at(i);
+    const auto value = mediaValue->data().at(i).toReal();
+
+    if (value >= m_maxValue)
+      m_maxValue = value;
+
+    if (value <= m_minValue)
+      m_minValue = value;
+
+    const auto point = QPointF(i, value);
+    m_linePoints.append(point);
+  }
+
+  m_color = jsonColorHelper(m_jsonColorArr, 0);
```

Comment:
These are much nicer. I did consider making the parameter `jsonColorHelper(m_jsonColorArr[index])` but we lose the index verification. I don't mind this function signature but there is the possibility to remove a parameter so I wanted to acknowledge that.

Reply:
That makes sense. I like it the way you have it, since the helper takes on the burden of validating the json at that index, which makes the calling code more readable.